x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/dimension_editor.tsx

```diff
@@ -306,24 +299,28 @@ export function DimensionEditor(props: DimensionEditorProps) {
               incompatibleSelectedOperationType={incompatibleSelectedOperationType}
               onDeleteColumn={() => {
                 setState(
-                  deleteColumn({
+                  mergeLayer({
                     state,
                     layerId,
-                    columnId,
+                    newLayer: deleteColumn({ layer: state.layers[layerId], columnId }),
                   })
                 );
               }}
               onChoose={(choice) => {
-                let column: IndexPatternColumn;
+                let newLayer: IndexPatternLayer;
                 if (
                   !incompatibleSelectedOperationType &&
                   selectedColumn &&
                   'field' in choice &&
                   choice.operationType === selectedColumn.operationType
                 ) {
-                  // If we just changed the field are not in an error state and the operation didn't change,
-                  // we use the operations onFieldChange method to calculate the new column.
-                  column = changeField(selectedColumn, currentIndexPattern, fieldMap[choice.field]);
+                  newLayer = replaceColumn({
+                    layer: state.layers[layerId],
+                    columnId,
+                    indexPattern: currentIndexPattern,
+                    field: fieldMap[choice.field],
+                    // TODO: don't keep params?
```

Comment:
Does this comment mean you'd prefer not to keep params?

Reply:
It seems like we should resolve this comment on the PR - I don't think we can avoid keeping params because it's necessary to carry over things like the number of values when switching from top values based on one field to top values based on another field